File: motomanager/backend/MotoManager.Domain/Entities/ServiceOrderOperation.cs
namespace MotoManager.Domain.Entities;

public class ServiceOrderOperation
{
    public long Id { get; set; }
    public long ServiceOrderId { get; set; }
    public long ServiceOperationId { get; set; }
    public decimal WorkHours { get; set; }
    public decimal PricePerHour { get; set; }
    public decimal TotalPrice { get; set; }

    public ServiceOrder? ServiceOrder { get; set; }
    public ServiceOperation? ServiceOperation { get; set; }
}

File: motomanager/frontend/motomanager-web/src/app/models/service-order-operation.ts
export interface ServiceOrderOperation {
  id: number;
  serviceOrderId: number;
  serviceOperationId: number;
  operationName: string;
  workHours: number;
  pricePerHour: number;
  totalPrice: number;
}

export interface AddServiceOrderOperationRequest {
  serviceOperationId: number;
  workHours: number;
  pricePerHour: number;
}

export interface UpdateServiceOrderOperationRequest {
  workHours: number;
  pricePerHour: number;
}
